src/test/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/MappingResourceIntTest.kt

```diff
@@ -0,0 +1,84 @@
+package uk.gov.justice.digital.hmpps.nomisvisitsmappingservice.resource
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.MediaType
+import org.springframework.web.reactive.function.BodyInserters
+import uk.gov.justice.digital.hmpps.nomisvisitsmappingservice.data.MappingDto
+import uk.gov.justice.digital.hmpps.nomisvisitsmappingservice.helper.builders.Repository
+import uk.gov.justice.digital.hmpps.nomisvisitsmappingservice.integration.IntegrationTestBase
+
+private const val nomisId = 1234L
+private const val vsipId = "12345678"
+
+private fun createMapping(): MappingDto {
+  return MappingDto(
+    nomisId = nomisId,
+    vsipId = vsipId,
+    label = "2022-01-01",
+    visitType = "ONLINE",
+  )
+}
+
+class MappingResourceIntTest : IntegrationTestBase() {
+
+  @Autowired
+  lateinit var repository: Repository
+
+  @DisplayName("Create")
+  @Nested
+  inner class CreateMappingTest {
+
+    @AfterEach
+    internal fun deleteData() {
+      repository.delete(nomisId)
+    }
+
+    @Test
+    fun `create mapping success`() {
+      webTestClient.post().uri("/mapping")
+        .headers(setAuthorisation(roles = listOf("ROLE_UPDATE_NOMIS")))
```

Comment:
would also be go to ensure that spring security is setup correct and unauthorised is returned if wrong role

Reply:
will sort the rest of the tests in next PR